pkg/services/preference/prefimpl/pref.go

```diff
@@ -247,3 +275,65 @@ func (s *Service) GetDefaults() *pref.Preference {
 func (s *Service) DeleteByUser(ctx context.Context, userID int64) error {
 	return s.store.DeleteByUser(ctx, userID)
 }
+
+func (s *Service) saveEntity(ctx context.Context, p *pref.Preference) error {
+	if s.entitystore == nil {
+		return nil
+	}
+
+	// Convert from pref.Preference to preferences
+	m := preferences.Preferences{}
+	if p.HomeDashboardID > 0 {
+		uid := ""
+		err := s.sess.Get(ctx, &uid, "SELECT uid FROM dashboard WHERE id=?", p.HomeDashboardID) // orgID already taken care of
```

Comment:
Don't call other tables from the preferences service - that's exactly the functionality that will get broken when the `dashboard` table is in a different underlying database from `preferences` (it might not seem likely, but that's exactly the situation we're enabling). You'll need to add the dashboards.DashboardService as a dependency to preferences and run GetDashboardByID from there.